tests/suites/test_suite_ecdh.function

```diff
@@ -150,8 +152,10 @@ void ecdh_exchange( int id )
 
     TEST_ASSERT( mbedtls_ecdh_calc_secret( &srv, &len, buf, 1000,
                                    &rnd_pseudo_rand, &rnd_info ) == 0 );
-    TEST_ASSERT( mbedtls_ecdh_calc_secret( &cli, &len, buf, 1000, NULL, NULL ) == 0 );
-    TEST_ASSERT( mbedtls_mpi_cmp_mpi( &srv.z, &cli.z ) == 0 );
+    TEST_ASSERT( mbedtls_ecdh_calc_secret( &cli, &res_len, res_buf, 1000,
+                                   NULL, NULL ) == 0 );
+    TEST_ASSERT( len == res_len );
+    TEST_ASSERT( memcmp( buf, res_buf, len ) == 0 );
```

Comment:
Did we have to change these asserts (no longer using `mbedtls_mpi_cmp_mpi()` because MPI might not be used in the underlying implementation? Did we change the format of data returned by `mbedtls_ecdh_calc_secret()` (whether in the default configuration or otherwise)?

Reply:
Yes, that was the reason. No, we didn't, it is unsigned fixed size integer in a big endian byte order like it was before.

Reply:
OK. So, again, the old way works with the legacy configuration still, and we've only changed this test to use the new way required so that it can use both the old and new configuration.